### ISSUE TITLE
No way to differentiate between regression and classification at the Booster level

### ISSUE THREAD

**Issue by calebmadrigal** [Microsoft/LightGBM#1700]
This is more of a feature request than a bug. But if I have a `lightgbm.basic.Booster` object, I see no way to determine if that object is a regressor or classifier. At the `LGBMClassifier`/`LGBMRegressor` level, the type itself clearly demarcates, but at the `Booster` level, I see nothing I can look at to determine which it is.

Perhaps I'm missing something, but if there indeed is no way to differentiate between classifier/regressor, I think it would be a good enhancement.

**Comment by guolinke** [comment]
I think this feature is easy to implement, just record the objective function or the types of labels, when initializing the booster.

**Comment by StrikerRUS** [comment]
Fixed in #1850.

Now `objective` can be easily retrieved at the booster level via `dump_model()`:
```
booster = lgb.train({'objective': 'binary'}, train_data)
booster.dump_model()['objective']
>>> 'binary sigmoid:1'


booster = lgb.train({}, train_data)
booster.dump_model()['objective']
>>> 'regression'
```

Thanks for pointing to this functionality omission, @calebmadrigal !
